File: backend/app/schemas/report.py
from __future__ import annotations

from pydantic import BaseModel
from typing import Optional


class ReportSection(BaseModel):
    title: str
    content: str


class AnalystReportResponse(BaseModel):
    ticker: str
    name: str
    generatedAt: str
    rating: str
    confidenceScore: float
    sector: Optional[str] = None
    analystName: str = "BlackGrid Research"
    sections: list[ReportSection]
    executiveSummary: str
    keyHighlights: str = ""
    technicalView: str
    fundamentalSnapshot: str
    macroContext: str
    forecastView: str
    valuationScenarios: str = ""
    competitiveLandscape: str = ""
    bullCase: str
    bearCase: str
    risksCatalysts: str
    analystConclusion: str = ""
    disclaimer: str = ""

File: types/report.ts
export interface ReportSection {
  title: string;
  content: string;
}

export interface ReportPricePoint {
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
}

export interface ReportQuote {
  price: number;
  change: number;
  changePercent: number;
  volume: number;
}

export interface AnalystReportResponse {
  ticker: string;
  name: string;
  generatedAt: string;
  rating: "Strong Buy" | "Buy" | "Hold" | "Sell" | "Strong Sell";
  confidenceScore: number;
  sector?: string;
  analystName: string;
  sections: ReportSection[];
  executiveSummary: string;
  keyHighlights: string;
  technicalView: string;
  fundamentalSnapshot: string;
  macroContext: string;
  forecastView: string;
  valuationScenarios: string;
  competitiveLandscape: string;
  bullCase: string;
  bearCase: string;
  risksCatalysts: string;
  analystConclusion: string;
  disclaimer: string;
  // Chart data and quote for rendering
  quote?: ReportQuote;
  chart?: ReportPricePoint[];
}
